src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/utils.h

```diff
@@ -0,0 +1,197 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+// See: https://spdx.org/licenses/
+
+#ifndef CORE_UTILS_H_
+#define CORE_UTILS_H_
+
+#if defined(GRPC_CHANNEL)
+#include <message_infrastructure/csrc/channel/grpc/grpcchannel.grpc.pb.h>
+#endif
+#include <message_infrastructure/csrc/core/message_infrastructure_logging.h>
+#include <memory>
+#include <chrono>  // NOLINT
+#include <thread>  // NOLINT
+
+#if defined(ENABLE_MM_PAUSE)
+#include <immintrin.h>
+#endif
+
+#define MAX_ARRAY_DIMS (5)
+#define SLEEP_NS (1)
+
+#define LAVA_SIZEOF_CHAR       (sizeof(char))
+#define LAVA_SIZEOF_UCHAR      (LAVA_SIZEOF_CHAR)
+
+#define LAVA_SIZEOF_BOOL       (LAVA_SIZEOF_UCHAR)
+#define LAVA_SIZEOF_BYTE       (LAVA_SIZEOF_CHAR)
+#define LAVA_SIZEOF_UBYTE      (LAVA_SIZEOF_UCHAR)
+#define LAVA_SIZEOF_SHORT      (sizeof(short))  // NOLINT
+#define LAVA_SIZEOF_USHORT     (sizeof(u_short))
+#define LAVA_SIZEOF_INT        (sizeof(int))
+#define LAVA_SIZEOF_UINT       (sizeof(uint))
+#define LAVA_SIZEOF_LONG       (sizeof(long))  // NOLINT
+#define LAVA_SIZEOF_ULONG      (sizeof(ulong))
+#define LAVA_SIZEOF_LONGLONG   (sizeof(long long))  // NOLINT
+#define LAVA_SIZEOF_ULONGLONG  (sizeof(ulong long))  // NOLINT
+#define LAVA_SIZEOF_FLOAT      (sizeof(float))
+#define LAVA_SIZEOF_DOUBLE     (sizeof(double))
+#define LAVA_SIZEOF_LONGDOUBLE (sizeof(long double))
+#define LAVA_SIZEOF_NULL       (0)
+// the length of string is unknown
+#define LAVA_SIZEOF_STRING     (-1)
+
+#define SIZEOF(TYPE) (LAVA_SIZEOF_ARRAY[TYPE])
+
+namespace message_infrastructure {
+
+enum ProcessType {
```

Comment:
For all enums, please use "enum class" to avoid namespace pollution and make it clear locally what the enum type is.

Also, avoid explicitly setting the values of the enums if their values are not important.